Utilities/General/scripts/ibeos-lfn-sort

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+ibeos_err(){
+	CMSSW_PREF=$1
+	echo "You have executed this command without the correct PATH setting. Please run the following and then rerun your command:"
+	echo 'PATH=$'$CMSSW_PREF':$PATH'
+	exit 1
+}
+
+# swallow piped input
+while read LFN ; do
+	:
+done
+
+CMSSW_PATH=src/Utilities/General/ibeos
+for CMSSW_VAR in CMSSW_BASE CMSSW_RELEASE_BASE; do
+	if [ -n ${!CMSSW_VAR} ] && [ -e ${!CMSSW_VAR}/${CMSSW_PATH} ]; then
+		ibeos_err ${CMSSW_VAR}/${CMSSW_PATH}
+	fi
+done
+
+echo "CMSSW environment not set"
+exit 1
```

Comment:
What is the code path this script exits with code `0`?

Reply:
it is under `cmssw/src/Utilities/General/ibeos` (see the for loop above)

Reply:
I still don't follow (maybe this just goes beyond my level of bash...). My understanding of the behavior is that
* If `$CMSSW_BASE` is non-empty, and `$CMSSW_BASE/src/Utilities/General/ibeos` exists, then
  * Call `ibeos_err` function with `CMSSW_BASE/src/Utilities/General/ibeos` argument
  * `ibeos_err` assigns `CMSSW_BASE/src/Utilities/General/ibeos` to `$CMSSW_PREF`, prints out the warning, and exits with `1`
* Same for `$CMSSW_RELEASE_BASE`
* If reach line 23, exit with `1`

Aha, I think I see point now. Is the point that the `${CMSSW_BASE,CMSSW_RELEASE_BASE}/src/Utilities/General/ibeos` should be prepended to the `$PATH` so that the `.../ibeos/ibeos-lfn-sort` is executed instead of this `.../scripts/ibeos-lfn-sort`? I think it would good to explain this (and the reason for this complexity) in comments in the beginning of this script.

Reply:
I added an explanation of this in a comment at the top of this script. (To confirm, this script should indeed always exit with code 1.)

Reply:
Thanks Kevin for the explanation, it makes the motivation of the setup clear.